instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java

```diff
@@ -169,51 +185,200 @@ void captureMessage(LogRecordBuilder builder, AttributesBuilder attributes, Mess
 
     if (captureMapMessageAttributes) {
       // TODO (trask) this could be optimized in 2.9 and later by calling MapMessage.forEach()
-      mapMessage
-          .getData()
-          .forEach(
-              (key, value) -> {
-                if (value != null
-                    && (!checkSpecialMapMessageAttribute
-                        || !key.equals(SPECIAL_MAP_MESSAGE_ATTRIBUTE))) {
-                  attributes.put(getMapMessageAttributeKey(key), value.toString());
-                }
-              });
+      consumeAttributes(
```

Comment:
previously all values were Strings now they are not. Isn't this a backwards incompatible change?

Reply:
4ff3198ec34ef60a30ffe9baee83f8f865049293

Reply:
Should this be hidden behind a feature flag?